switch to PlaneBufferGeometry

morandd opened the pull request:
Big speedup in initialization time by switching from THREE.PlaneGeometry to Three.PlaneBufferGeometry. Saves about 3 seconds on init time in my case.

Commit:
switch to PlaneBufferGeometry

bryik commented on index.js:
```diff
@@ -48,8 +48,9 @@ AFRAME.registerComponent('terrain-model', {
    */
-  multiple: false,
+  multiple: true,
 
```
This change is fine. It was originally `false` because the component boilerplate has it set to `false` and I had not changed it.

bryik commented on index.js:
```diff
@@ -71,25 +72,27 @@ AFRAME.registerComponent('terrain-model', {
+        verts.array[vi] = data[di] / 65535 * zPosition;
       }
+      geometry.attributes.position.needsUpdate = true;
 
```
This for loop is getting a bit hairy. Would you mind adding a comment so I'll be able to remember what these variables mean? Something like:

```js
/**
 * vi = The index of the current vertex's Z position in the plane geometry's position attribute array.
 * di = The index of the current data value.
 * l = The total number of vertices in the plane geometry.
 */
```

morandd commented:
I went fairly far into this and in the end so much changed that I think it's better to have a separate component. This component is great for actual ENVI based data, whereas my component is designed more for data vis, with options to suit that. 

Maybe you could mention my component in your README?
https://github.com/morandd/aframe-heatmap3d

I have linked to this component and acknowledged that my work is 110% based on your (and the underlying) work. Thanks for getting this started!

bryik commented:
I've added a link to your component in my readme. I might still make the change to PlaneBufferGeometry though, better perf for little cost. Thanks for notifying me about it :)

morandd commented:
cool -- that key 'hairy' for loop is the same in heapmap3d but I cleaned up the variable names a bit, so maybe take a look there. 

I hope it's ok that I split that one off. but as you see, the input parameters and internals are rather different so adding all those features on top of yours would've really complicated it.

bryik commented:
Yeah looking at your component, I can understand why you split it off from mine (it's quite different). This is one of the great things about open source--the freedom to remix!
